R/estimate_functions.R

```diff
@@ -20,48 +19,72 @@ get_kappa_grid <- function(method, n_kappa, kappa) {
 }
 
 #' @importFrom lpSolve lp
-is_feasible <- function(X, Y) {
+#checks if MLE exists
+is_feasible <- function(X, Y) { 
   solution_object <- lp("min", 0, X, 
                         ifelse(Y == 0, "<=", ">="), 
-                        ifelse(Y == 0, -1, 1)) #weak inequality
-  solution_object[28] == 0
+                        ifelse(Y == 0, -1, 1)) 
+  #we do not put zero because of weak inequality
+  solution_object[28] == 0 
+  #if true, the problem is feasible, so MLE does not exist
 }
 
-estimate_pi <- function(X, Y, n_rep, kappa) {
+#caclulates how often MLE doesn't exist for i-th kappa
+estimate_pi_i <- function(X, Y, n_rep, kappa_i) { 
+  n_variables <- ncol(X)
+  n_observations <- nrow(X)
   feasibile <- sapply(1:n_rep, function(i) {
-    ind <- sample(1:nrow(X), round(ncol(Y)/kappa))
+    ind <- sample(1:n_observations, round(n_variables/kappa_i)) 
     is_feasible(X[ind, ], Y[ind])
   })
   sum(feasibile)/n_rep
+  print("hello world")
 }
 
-estimate_pi_kappa <- function(X, Y, n_rep, n_observations, n_variables, kappa_grid) {
-  lapply(kappa_grid, function(kappa) {
-    estimate_pi(X, Y, n_rep, kappa)
+#caluculates pi_hats for kappa grid
+estimate_pi_kappagrid <- function(X, Y, n_rep, kappa_grid) {
+  n_variables <- ncol(X)
+  n_observations <- nrow(X)
+  sapply(kappa_grid, function(kappa_i) {
+    estimate_pi_i(X, Y, n_rep, kappa_i)
   })
 }
```

Comment:
Shouldn't this function return a single value (estimate of pi) instead of a vector?
Btw check if `vapply` can replace sapply in your code
Also, there might be something wrong with the name here

Reply:
You mean single value and only then apply this function on the grid?
I will check this.
Is it too long or does it not fit if function returns a single value?